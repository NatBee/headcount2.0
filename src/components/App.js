import React, { Component } from 'react';
import Controls from './Controls';
import ComparisonContainer from './ComparisonContainer';
import CardContainer from './CardContainer';
import DistrictRepository from './helper';
import '../styles/App.css';
import kinderData from '../data/kindergartners_in_full_day_program.js';

const DistrictData = new DistrictRepository(kinderData);

class App extends Component {

  constructor() {
    super()
    this.state = {
      allDistrictData: [],
      districtCards: []
    }
  }

  componentDidMount() {
     this.setState({
      allDistrictData: DistrictData.data
    })
  }

  handleSearch = (district) => {
    this.setState({
      districtCards: DistrictData.findByName(district)
    })
  }

  render() {
    return (
      <section>
        <Controls />
        <ComparisonContainer />
        <CardContainer allDistrictData={ this.state.allDistrictData }/>
      </section>
    );
  }
}

export default App;
