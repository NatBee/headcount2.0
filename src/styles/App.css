
.toggle-compare{
  position: absolute;
  z-index: 100;
  left: 0;
  top: 18%;
  font-size: 20px;
  width: 200px;
  line-height: 45px;
  border-top-right-radius: 10px;
}

.toggle-compare.hide{
  display: none;
}

.toggle-compare.show{
  display: block;
}

.toggle-compare img{
  transform: rotate(90deg);
  width: 25px;
  margin: 5px 0 -6px 10px;
}


.comparison-title{
  color: white;
}