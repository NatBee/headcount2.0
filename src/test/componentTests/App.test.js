import React from 'react';
import ReactDOM from 'react-dom';
import App from '../../components/App';
import { shallow } from 'enzyme';

describe('App tests', () => {
  let renderedComponent;

  beforeEach(() => {
    renderedComponent = shallow(<App/>)

    const mockedDistrict = {"data": {"2004": 0.24, 
                            "2005": 0.278, 
                            "2006": 0.337, 
                            "2007": 0.395, 
                            "2008": 0.536, 
                            "2009": 0.598, 
                            "2010": 0.64, 
                            "2011": 0.672, 
                            "2012": 0.695, 
                            "2013": 0.703, 
                            "2014": 0.741}, 
                            "location": "COLORADO"}
  })
  
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<App />, div);
  });

  it('should match the snapshot', () => {
    expect(renderedComponent).toMatchSnapshot()
  })

  it('should gather clean data from all districts', () => {
    expect(Object.keys(renderedComponent.state().allDistrictData).length).toBe(181)
  });

  it('when handleSearch is called with a district, the state should have that district object', () => {
    const mockedDistrict = {"data": {"2004": 0.24, 
                            "2005": 0.278, 
                            "2006": 0.337, 
                            "2007": 0.395, 
                            "2008": 0.536, 
                            "2009": 0.598, 
                            "2010": 0.64, 
                            "2011": 0.672, 
                            "2012": 0.695, 
                            "2013": 0.703, 
                            "2014": 0.741}, 
                            "location": "COLORADO"}
  

    renderedComponent.instance().handleSearch('COLORADO')
    expect(renderedComponent.state().allDistrictData[0]).toEqual(mockedDistrict)
  })

  it('should have 1 card in cardCompData when compare care function runs once', () => {
    renderedComponent.instance().compareCards('COLORADO')
    expect(renderedComponent.state().cardCompData.length).toEqual(1) 
  })

  it('should have 2 cards in cardCompData array', () => {
    renderedComponent.instance().compareCards('COLORADO')
    renderedComponent.instance().compareCards('COLORADO SPRINGS 11')
    expect(renderedComponent.state().cardCompData.length).toEqual(2)

  })

  it('should compare the cards when there are 2', () => {
    const mockedCompareData = 
      {data: {2004: 0.24, 2005: 0.278, 2006: 0.337, 2007: 0.395, 2008: 0.536, 2009: 0.598, 2010: 0.64, 2011: 0.672, 2012: 0.695, 2013: 0.703, 2014: 0.741}
location: "COLORADO"}, 
{data: {2004: 0.069, 2005: 0.509, 2006: 0.638, 2007: 0.994, 2008: 0.992, 2009: 1, 2010: 0.993, 2011: 0.994, 2012: 0.993, 2013: 0.989, 2014: 0.994}
location: "COLORADO SPRINGS 11"};

    renderedComponent.instance().comparison(['COLORADO', 'COLORADO SPRINGS 11'])
    console.log(renderedComponent.state().comparisonData)
    expect(renderedComponent.state().comparisonData).toEqual(mockedCompareData)
  })

  it('it should remove a card from cardCompData array', () => {
    const mockedCompareData = {
      {data: {2004: 0.24, 2005: 0.278, 2006: 0.337, 2007: 0.395, 2008: 0.536, 2009: 0.598, 2010: 0.64, 2011: 0.672, 2012: 0.695, 2013: 0.703, 2014: 0.741}
location: "COLORADO"}
    renderedComponent.instance().removeComparisonCard('COLORADO SPRINGS 11')
    expect(renderedComponent.state().comparisonData).toEqual(mockedCompareData)
  })

})



