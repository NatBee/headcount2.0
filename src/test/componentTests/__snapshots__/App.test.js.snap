// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Controls
          handleSearch={[Function]}
/>,
        <ComparisonContainer
          cardCompData={Array []}
          comparisonData={Object {}}
          removeComparisonCard={[Function]}
/>,
        <CardContainer
          allDistrictData={Array []}
          compareCards={[Function]}
/>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "handleSearch": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "cardCompData": Array [],
          "comparisonData": Object {},
          "removeComparisonCard": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "allDistrictData": Array [],
          "compareCards": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Controls
            handleSearch={[Function]}
/>,
          <ComparisonContainer
            cardCompData={Array []}
            comparisonData={Object {}}
            removeComparisonCard={[Function]}
/>,
          <CardContainer
            allDistrictData={Array []}
            compareCards={[Function]}
/>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "handleSearch": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "cardCompData": Array [],
            "comparisonData": Object {},
            "removeComparisonCard": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "allDistrictData": Array [],
            "compareCards": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
